### DISCUSSION TITLE
Is OIDCClientID a singleton?

### DISCUSSION THREAD

**Discussion by christian-2** [OpenIDC/mod_auth_openidc#1070] [Q&A] [answered]
I have noticed that placing `OIDCClientID` inside a `<Directory>` or `<Location>` context raises an error when the configuration is tested (e.g. "OIDCClientID not allowed in <Directory> context"). 

Does this mean that Apache can only aware of a single OIDC client ID (via `auth_openidc_module`), and if I wanted to connect to different OIDC clients I would have to run multiple instances of Apache?

**Comment by zandbelt** [comment, marked as answer]
that is correct, multiple vhosts may be a way around that; in general it should not be necessary to use different clients - in the same security realm that Apache/mod_auth_openidc is - as the same goal should be achievable through scopes

**Reply by christian-2** [reply]
For context: I am working on a demo that involves different OIDC clients with different characteristics. So I will switch to multiple instances of Apache. It's not a big deal on my side. Thanks for the quick response.
